core/test/element_math_test.cpp

```diff
@@ -24,34 +24,9 @@ TEST(ElementAbsTest, value_and_variance) {
   EXPECT_EQ(element::abs(x), abs(x));
 }
 
-TEST(ElementAbsOutArgTest, unit) {
-  units::Unit m(units::m);
-  units::Unit out(units::one);
-  element::abs_out_arg(out, m);
-  EXPECT_EQ(out, units::abs(m));
-}
-
-TEST(ElementAbsOutArgTest, value_double) {
-  double out;
-  element::abs_out_arg(out, -1.23);
-  EXPECT_EQ(out, std::abs(-1.23));
-}
-
-TEST(ElementAbsOutArgTest, value_float) {
-  float out;
-  element::abs_out_arg(out, -1.23456789f);
-  EXPECT_EQ(out, std::abs(-1.23456789f));
-}
-
-TEST(ElementAbsOutArgTest, value_and_variance) {
-  const ValueAndVariance x(-2.0, 1.0);
-  ValueAndVariance out(x);
-  element::abs_out_arg(out, x);
-  EXPECT_EQ(out, abs(x));
-}
-
-TEST(ElementAbsOutArgTest, supported_types) {
-  auto supported = decltype(element::abs_out_arg)::types{};
+TEST(ElementAbsTest, supported_types) {
+  auto supported = decltype(element::abs)::types{};
+  static_assert(std::tuple_size_v<decltype(supported)> == 2);
```

Comment:
Is this really a requirement? Will need to change this when adding new types.

Reply:
The idea is to ensure that if someone adds support for new types, this ensures that they at least look at the test and hopefully add tests for the new type.

Reply:
I think that makes sense. But on the other hand, if you look at things long term, unit tests are often interpreted as documenting requirements. So in 3 years someone might look at this and wonder if anything will break if support for new types is added, since apparently someone required support of exactly 2.

Reply:
Would a comment be enough to avoid that?

Reply:
Partially yes, but we should still ask ourselves what we gain.

Put differently: I think I generally strive (but often fail) to write tests that test the *contract*. Any other tests will sooner or later get in the way when refactoring, or cause confusion. In this case, supporting only 2 types is not in the contract.

In this case maybe we should weigh the risk of things breaking when a new types is added (but not a test), vs. hurdles when changing things. I am not sure what the answer is in this case.

Reply:
I think I get what you mean. Given that you probably wrote the rest of the tests in the way you describe, I think we should revert this test.